contracts/examples/ERC20Verifier.sol

```diff
@@ -1,29 +1,39 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.16;
+pragma solidity 0.8.20;
 
-import {ERC20} from '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import {ERC20Upgradeable} from '@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol';
 import {PrimitiveTypeUtils} from '@iden3/contracts/lib/PrimitiveTypeUtils.sol';
 import {ICircuitValidator} from '@iden3/contracts/interfaces/ICircuitValidator.sol';
 import {ZKPVerifier} from '@iden3/contracts/verifiers/ZKPVerifier.sol';
 
-contract ERC20Verifier is ERC20, ZKPVerifier {
+contract ERC20Verifier is ERC20Upgradeable, ZKPVerifier {
     uint64 public constant TRANSFER_REQUEST_ID_SIG_VALIDATOR = 1;
     uint64 public constant TRANSFER_REQUEST_ID_MTP_VALIDATOR = 2;
 
     mapping(uint256 => address) public idToAddress;
     mapping(address => uint256) public addressToId;
 
-    uint256 public TOKEN_AMOUNT_FOR_AIRDROP_PER_ID = 5 * 10**uint256(decimals());
+    uint256 public TOKEN_AMOUNT_FOR_AIRDROP_PER_ID;
 
-    constructor(string memory name_, string memory symbol_) ERC20(name_, symbol_) {}
+    function initialize(
+        string memory name,
+        string memory symbol,
+        address initialOwner
```

Comment:
I think we can omit this parameter for simplification and use `_msgSender()` for the `__ZKPVerifier_init()` method instead.

Reply:
removed